lldb/tools/lldb-dap/Protocol/ProtocolRequests.h

```diff
@@ -54,6 +54,79 @@ bool fromJSON(const llvm::json::Value &, DisconnectArguments &,
 /// body field is required.
 using DisconnectResponse = VoidResponse;
 
+/// Arguments for `initialize` request.
+struct InitializeRequestArguments {
+  /// The ID of the debug adapter.
+  std::string adatperID;
+
+  /// The ID of the client using this adapter.
+  std::optional<std::string> clientID;
+
+  /// The human-readable name of the client using this adapter.
+  std::optional<std::string> clientName;
+
+  /// The ISO-639 locale of the client using this adapter, e.g. en-US or de-CH.
+  std::optional<std::string> locale;
+
+  enum class PathFormat { path, uri };
+
+  /// Determines in what format paths are specified. The default is `path`,
+  /// which is the native format.
+  std::optional<PathFormat> pathFormat = PathFormat::path;
+
+  /// If true all line numbers are 1-based (default).
+  std::optional<bool> linesStartAt1;
+
+  /// If true all column numbers are 1-based (default).
+  std::optional<bool> columnsStartAt1;
+
+  enum class Feature {
+    /// Client supports the `type` attribute for variables.
+    supportsVariableType,
+    /// Client supports the paging of variables.
+    supportsVariablePaging,
+    /// Client supports the `runInTerminal` request.
+    supportsRunInTerminalRequest,
+    /// Client supports memory references.
+    supportsMemoryReferences,
+    /// Client supports progress reporting.
+    supportsProgressReporting,
+    /// Client supports the `invalidated` event.
+    supportsInvalidatedEvent,
+    /// Client supports the `memory` event.
```

Comment:
I don't mind re-adding those comments, but I omitted them in my earlier patch because they're pretty much adding no value. I think `supportsMemoryEvent` and `supportsANSIStyling` are the only two that deserve a comment.

Reply:
I trimmed most of the comments.